### ISSUE TITLE
dbeaver unexpect close due to invalid pointer

### ISSUE THREAD

**Issue by hubutui** [dbeaver/dbeaver#12422]
#### System information: 
- ArchLinux
- dbeaver 21.0.4
- openjdk 15.0.2 2021-01-19, OpenJDK Runtime Environment (build 15.0.2+7), OpenJDK 64-Bit Server VM (build 15.0.2+7, mixed mode)

#### Connection specification:
- Database name and version: MySQL and Oracle
- Do you use tunnels or proxies (SSH, SOCKS, etc)? : No

#### Describe the problem you're observing:

When I edit a connection, and click OK, dbeaver freezes with no response, and then close.

#### Include any warning/errors/backtraces from the logs

log file of command `LANG=en_US.UTF-8 dbeaver`: [dbeaver.log](https://github.com/dbeaver/dbeaver/files/6462964/dbeaver.log)
and `~/.local/share/DBeaverData/workspace6/.metadata/dbeaver-debug.log`: 
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/6462968/dbeaver-debug.log), and the jvm log: [hs_err_pid220752.log](https://github.com/dbeaver/dbeaver/files/6462972/hs_err_pid220752.log)

**Comment by ShadelessFox** [comment]
Looks like a GTK error.

Can you provide more information about your environment, please?

**Comment by hubutui** [comment]
@ShadelessFox Yes, I'm using ArchLinux with GNOME 40.1, gtk3 3.24.29. What else information should I provide?

**Comment by ShadelessFox** [comment]
That's enough for now, thank you. 

We need further investigation, at least have a try to reproduce it using the same environment.

One more question, Is that an one-time situation or it keeps happening every time you're trying to perform described actions?

**Comment by hubutui** [comment]
It keeps happening almost every time I trying to edit the connection setting.
